core/src/main/java/org/elasticsearch/search/rescore/RescoreBuilder.java

```diff
@@ -19,256 +19,20 @@
 
 package org.elasticsearch.search.rescore;
 
-import org.elasticsearch.ExceptionsHelper;
 import org.elasticsearch.common.io.stream.NamedWriteable;
-import org.elasticsearch.common.io.stream.StreamInput;
-import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.common.xcontent.ToXContent;
-import org.elasticsearch.common.xcontent.XContentBuilder;
-import org.elasticsearch.common.xcontent.XContentFactory;
-import org.elasticsearch.index.query.MatchAllQueryBuilder;
-import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryParseContext;
+import org.elasticsearch.index.query.QueryShardContext;
 
 import java.io.IOException;
-import java.util.Locale;
-import java.util.Objects;
 
-public class RescoreBuilder implements ToXContent, Writeable<RescoreBuilder> {
+public interface RescoreBuilder<RB extends RescoreBuilder<?>> extends ToXContent, NamedWriteable<RB> {
```

Comment:
I don't think we need it, but thats the way we also do it with QueryBuilders (Interface, Abstract base class). I wonder if we should change the name of the abstract class to RescoreBuilder then, that would be similar to HighlightBuilder and ShapeBuilders then, wdyt?